src/Simplify_Div.cpp

```diff
@@ -3,112 +3,45 @@
-        const bool b_negative = b_bounds.max_defined && b_bounds.max < 0;
-
-        if ((b_positive && !b_bounds.max_defined) ||
-            (b_negative && !b_bounds.min_defined)) {
-            // Take limit as b -> +/- infinity
-            int64_t v = 0;
-            bounds->min = std::min(bounds->min, v);
-            bounds->max = std::max(bounds->max, v);
-        }
-
-        bounds->min_defined = ((a_bounds.min_defined && b_positive) ||
-                               (a_bounds.max_defined && b_negative));
-        bounds->max_defined = ((a_bounds.max_defined && b_positive) ||
-                               (a_bounds.min_defined && b_negative));
-
-        // That's as far as we can get knowing the sign of the
-        // denominator. For bounded numerators, we additionally know
-        // that div can't make anything larger in magnitude, so we can
-        // take the intersection with that.
-        if (a_bounds.max_defined && a_bounds.min_defined) {
-            int64_t v = std::max(a_bounds.max, -a_bounds.min);
-            if (bounds->min_defined) {
-                bounds->min = std::max(bounds->min, -v);
-            } else {
-                bounds->min = -v;
-            }
-            if (bounds->max_defined) {
-                bounds->max = std::min(bounds->max, v);
-            } else {
-                bounds->max = v;
-            }
-            bounds->min_defined = bounds->max_defined = true;
-        }
-
-        // Bounded numerator divided by constantish
-        // denominator can sometimes collapse things to a
-        // constant at this point
-        if (bounds->min_defined &&
-            bounds->max_defined &&
-            bounds->max == bounds->min) {
-            if (op->type.can_represent(bounds->min)) {
-                return make_const(op->type, bounds->min);
-            } else {
-                // Even though this is 'no-overflow-int', if the result
-                // we calculate can't fit into the destination type,
-                // we're better off returning an overflow condition than
-                // a known-wrong value. (Note that no_overflow_int() should
-                // only be true for signed integers.)
-                internal_assert(op->type.is_int());
-                clear_bounds_info(bounds);
-                return make_signed_integer_overflow(op->type);
+Expr Simplify::visit(const Div *op, ExprInfo *info) {
+    ExprInfo a_info, b_info;
+    Expr a = mutate(op->a, &a_info);
+    Expr b = mutate(op->b, &b_info);
+
+    if (info) {
+        if (op->type.is_int_or_uint()) {
+            info->bounds = a_info.bounds / b_info.bounds;
+            info->alignment = a_info.alignment / b_info.alignment;
+            info->trim_bounds_using_alignment();
+            info->cast_to(op->type);
+
+            // Bounded numerator divided by constantish
+            // denominator can sometimes collapse things to a
+            // constant at this point
```

Comment:
What is "constantish"?

Reply:
This is just code motion (increased indent level), but there are cases where the denominator doesn't strictly need to be a constant that this handles, e.g. [3,9] / [15,23] = 0

Reply:
I think it would be useful to clarify that in the comment, maybe just adding that example